### ISSUE TITLE
neEffdecomp fails when covariate name is substring of outcome name

### ISSUE THREAD

**Issue by grmurchison** [jmpsteen/medflex#13]
In version 0.6-4, using models created with neImpute:

```
# New column PB with data from gender column and name a substring of the outcome variable
UPBdata$PB <- UPBdata$gender

# Single covariate with name a substring of the outcome
impData <- neImpute(UPB ~ att + negaff + PB, family = binomial, data = UPBdata)
neMod <- neModel(UPB ~ att0 + att1 + PB, family = binomial, expData = impData, se = "robust")
neEffdecomp(neMod)
```
> Error in dimnames(x) <- dn : length of 'dimnames' [2] not equal to array extent

```
# Two covariates, one with name a substring of the outcome
impData <- neImpute(UPB ~ att + negaff + PB + educ, family = binomial, data = UPBdata)
neMod <- neModel(UPB ~ att0 + att1 + PB + educ, family = binomial, expData = impData, se = "robust")
neEffdecomp(neMod)
```
> Error in varterms[sapply(cov, grep, varterms)] : invalid subscript type 'list'

However, the following model (same data, different names) succeeds:
```
# Covariate name not a substring of the outcome name (same data) 
impData <- neImpute(UPB ~ att + negaff + gender, family = binomial, data = UPBdata)
neMod <- neModel(UPB ~ att0 + att1 + gender, family = binomial, expData = impData, se = "robust")
neEffdecomp(neMod)
```

The problem appears to be in the following line of neEffdecomp:
`covTerms <- if (!is.null(cov)) varterms[sapply(cov, grep, varterms)] else NULL`
'grep' returns both the covariate and the outcome variable, when only the covariate should be returned.

**Comment by jmpsteen** [comment]
This should be solved in version 0.6-7 (which I just updated to github). Can you please verify and confirm? Thanks!

**Comment by grmurchison** [comment]
Yes, it is working fine in the new version. Thanks!
